mitmproxy/protocol/http.py

```diff
@@ -196,11 +195,16 @@ def __call__(self):
                     flow = self.channel.ask("responseheaders", flow)
                     if flow == Kill:
                         raise Kill()
+                    if flow is None:
+                        return
 
                 self.log("response", "debug", [repr(flow.response)])
                 flow = self.channel.ask("response", flow)
                 if flow == Kill:
                     raise Kill()
+                if flow is None or flow.response is None:
+                    return
```

Comment:
Similar to above, if h2 gets canceled there might be no response, or even no flow at all.
Yes, this is all very strange and dirty - but apparently happens (although not deterministically reproducible)